docs/paper/dev/api/event-api/chat-event.md

```diff
@@ -0,0 +1,156 @@
+        // ...
+    }
+
+}
+```
+
+To be able to use this renderer, we need to tell the event to use it.
+
+```java title="ChatEvent.java"
+public class ChatEvent implements Listener {
+
+    @EventHandler
+    public void onChat(AsyncChatEvent event) {
+        event.renderer(new ExampleRenderer());
+    }
+
+}
+```
+
+### Using a lambda expression
+
+Another way of using the renderer is by using a lambda expression.
+
+```java title="ChatEvent.java"
+public class ChatEvent implements Listener {
+
+    @EventHandler
+    public void onChat(AsyncChatEvent event) {
+        event.renderer((source, sourceDisplayName, message, viewer) -> {
+            // ...
+        });
+    }
+
+}
+```
+
+## Rendering the message
+
+Now that we have our renderer, we can start rendering the message.
+
+Let's say we want to render our chat to look like this:
+
+![](./assets/plain-message-rendering.png)
+
+To do this, we need to return a new `Component` that contains the message we want to send.
+
+```java title="ExampleRenderer.java"
+public class ExampleRenderer implements ChatRenderer {
+
+    @Override
+    public @NotNull Component render(@NotNull Player source, @NotNull Component sourceDisplayName, @NotNull Component message, @NotNull Audience viewer) {
+        return sourceDisplayName 
+                .append(Component.text(": ")) 
+                .append(message); 
+    }
+
+}
+```
+
+Now you can see that the message is rendered as we wanted it to be.
+
+## Rendering the message with a prefix using LuckPerms and MiniMessage
+
+Usually, you would want to add a prefix to the message. 
+In this example, we will be using LuckPerms to get the players prefix.
+Let's say we want to render our chat to look like this:
+
+![img.png](assets/prefix-rendering.png)
+
+This process is a bit more complex than the previous one.
+For this we create our `admin` group and add a prefix to it:
+
+- /lp creategroup admin
+
+- /lp group admin meta addprefix 100 "`<gray>[</gray><red>Admin</red><gray>]</gray>`"
+
+To get the prefix of the player, we need to use the `User` class from LuckPerms to access the cached data of the player.
+We can then use the `MetaData` class to get the prefix of the player and deserialize it using `MiniMessage`.
+
+::: info
+
+`MiniMessage` is a library that allows you to serialize and deserialize `Component` to and from a string using 
+readable html tags. You can read more about it [here](https://docs.advntr.dev/minimessage.html).
+
+:::
+
+```java title="ExampleRenderer.java"
+public class ExampleRenderer implements ChatRenderer {
+
+    @Override
+    public @NotNull Component render(@NotNull Player source, @NotNull Component sourceDisplayName, @NotNull Component message, @NotNull Audience viewer) {
+        User user = LuckPermsProvider.get().getPlayerAdapter(Player.class).getUser(source);
+        String prefixString = user.getCachedData().getMetaData().getPrefix();
+        Component prefix = MiniMessage.miniMessage().deserialize(prefixString);
```

Comment:
This will throw if the prefix is null, also if we're going to use LuckPerms as an example, might want to link to the api docs for the LP API?